src/foam/dao/AbstractDAO.java

```diff
@@ -58,6 +58,8 @@ protected Sink decorateSink_(Sink sink, long skip, long limit, Comparator order,
       sink = new PredicatedSink(predicate, sink);
     }
 
+    ((AbstractFObject) sink).setX(x);
```

Comment:
Went back if forth in my head for a while.  I get what you're saying.  Although in this context people couldn't redefine EQ() to have any ill effect since your PredicatedDAOs are already created, but they could redefine PredicatedSink.

In general I think things like DAOs which are both Context Aware and Context Oriented, should use the Context-Aware context for creating internal things, and not their context oriented context, which my suggestion would not be doing.

In this particular case, I think we should remove setX(), and also update OrderedSink to use "new foam.dao.Subscription()" rather than getX().create(foam.dao.Subscription.class); and everything should work for now.